src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -188,9 +190,19 @@ private CommandResult executeCommand(String commandText) throws CommandException
 
             return commandResult;
         } catch (CommandException | ParseException e) {
-            logger.info("Invalid command: " + commandText);
-            resultDisplay.setFeedbackToUser(e.getMessage());
+            if (e.getCause() instanceof AccessDeniedException) {
+                handleAccessDeniedException(e);
+            } else {
+                resultDisplay.setFeedbackToUser(e.getMessage());
+                logger.info("Invalid command: " + commandText);
+            }
```

Comment:
See if you can get rid of this condition (checking of object type should be avoided if possible). That is, if the correct error message was put into the exception in the first place, you don't need a different behaviors here. That means we only need to change the place where the error is detected.